sdks/wasm/Mono.WebAssembly.DebuggerProxy/WsClient.cs

```diff
@@ -0,0 +1,129 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+
+using System.Net.WebSockets;
+using System.Threading;
+using System.IO;
+using System.Text;
+using System.Collections.Generic;
+
+namespace Mono.WebAssembly {
+	public class WsClient: IDisposable {
+		ClientWebSocket socket;
+		List<Task> pending_ops = new List<Task> ();
+		TaskCompletionSource<bool> side_exit = new TaskCompletionSource<bool> ();
+		List<byte []> pending_writes = new List<byte []> ();
+		Task current_write;
+
+		public WsClient () {
+		}
+
+		~WsClient() {
+			Dispose(false);
+		}
+
+		public void Dispose() {
+			Dispose(true);
+		}
+
+		protected virtual void Dispose (bool disposing) {
+			if (disposing)
+				socket.Dispose ();
+		}
+
+		Task Pump (Task task, CancellationToken token)
+		{
+			if (task != current_write)
+				return null;
+			current_write = null;
+
+			pending_writes.RemoveAt (0);
+
+			if (pending_writes.Count > 0) {
+				current_write = socket.SendAsync (new ArraySegment<byte> (pending_writes [0]), WebSocketMessageType.Text, true, token);
+				return current_write;
+			}
+			return null;
+		}
+
+		async Task<string> ReadOne (CancellationToken token)
+		{
+			byte [] buff = new byte [4000];
```

Comment:
I'd make this bigger

Reply:
The vast majority of messages are < 100 bytes. If anything it should be using buffer polling.